presto-redshift/src/main/java/io/prestosql/plugin/redshift/RedshiftClient.java

```diff
@@ -62,6 +68,77 @@ protected void renameTable(JdbcIdentity identity, String catalogName, String sch
         }
     }
 
+    static String generateCreateTableSql(String fullyQualifiedTableName, ImmutableList.Builder<String> columnList, Optional<String> distKey, Optional<String> compoundSortKeys, Optional<String> interleavedSortKeys){
+        checkArgument(!(compoundSortKeys.isPresent() && interleavedSortKeys.isPresent()), "Cannot set both a compound sortkey and an interleaved sortkey in a Redshift table");
+
+        String distKeyString = distKey.map(k -> format("distkey(%s) ", k)).orElse("");
+        String compoundSortKeysString = compoundSortKeys.map(k -> format("compound sortkey(%s) ", k)).orElse("");
+        String interleavedSortKeysString = interleavedSortKeys.map(k -> format("interleaved sortkey(%s) ", k)).orElse("");
+
+        return format(
+                "CREATE TABLE %s (%s) %s%s%s",
+                fullyQualifiedTableName,
+                join(", ", columnList.build()),
+                distKeyString,
+                compoundSortKeysString,
+                interleavedSortKeysString
+        );
+    }
+
+    @Override
+    protected JdbcOutputTableHandle createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, String tableName)
```

Comment:
Most of this is copy pasted from the base jdbc implementation but I copy pasted it so that the base jdbc implementation wouldn't have redshift-specific logic in it.